packages/actions-server/src/ActionsProxy.test.ts

```diff
@@ -0,0 +1,279 @@
+import { describe, it, expect, beforeEach, afterEach } from "vitest";
+import { ActionsProxy } from "./ActionsProxy.js";
+import { Client as McpClient } from "@modelcontextprotocol/sdk/client/index.js";
+import { InMemoryTransport } from "@modelcontextprotocol/sdk/inMemory.js";
+import { z } from "zod";
+import { WebSocket } from "ws";
+
+interface ToolResult {
+    content: Array<{
+        type: "text";
+        text: string;
+    }>;
+    _meta: Record<string, unknown>;
+    isError: boolean;
+}
+
+interface ToolRequest {
+    type: 'TOOL_REQUEST';
+    payload: {
+        toolName: string;
+        parameters: Record<string, unknown>;
+        requestId: string;
+    };
+}
+
+describe("ActionsProxy", () => {
+    let server: ActionsProxy;
+    let client: McpClient;
+    let clientId: string;
+    let wsPort: number;
+
+    beforeEach(async () => {
+        // Use a random port between 30000-40000
+        wsPort = Math.floor(Math.random() * 10000) + 30000;
+        server = new ActionsProxy(wsPort);
+
+        // Register tools before connecting
+        clientId = await server.registerTools([
+            {
+                name: "testTool",
+                description: "A test tool",
+                parameters: z.object({
+                    input: z.string()
+                })
+            }
+        ]);
```

Comment:
This will be issued by a client side component that's got access to the client side MCP tools (in our case the `ActionsManager`)